auacm/angular/src/app/competition.service.ts

```diff
@@ -0,0 +1,341 @@
+              if (data[competitionType][i]['registered'] !== undefined) competition['registered'] = data[competitionType][i]['registered'];
+              if (data[competitionType][i]['startTime'] !== undefined) competition['startTime'] = data[competitionType][i]['startTime'];
+              if (data[competitionType][i]['stop'] !== undefined) competition['stop'] = data[competitionType][i]['stop'];
+              if (data[competitionType][i]['compProblems'] !== undefined) competition['compProblems'] = data[competitionType][i]['compProblems'];
+              competitions[competitionType].push(competition);
+            }
+          }
+        }
+        resolve(competitions);
+      }, (err: Response) => {
+        var competitions = new Map<string, Competition[]>();
+        competitions['ongoing'] = [];
+        competitions['upcoming'] = [];
+        competitions['past'] = [];
+        resolve(competitions);
+      });
+    })
+  };
+
+  getCompetition(cid: number) : Promise<Competition> {
+    return new Promise((resolve, reject) => {
+      this._http.get(`/api/competitions/${cid}`).subscribe((res: Response) => {
+        if (res.status === 200) {
+          var competition = new Competition();
+          var data = res.json().data;
+
+          // Parse competition compProblems
+          for (var key in data.compProblems) {
+            var tempData = data.compProblems[key];
+            var tempCompProblem = new CompetitionProblem();
+            tempCompProblem.name = tempData.name;
+            tempCompProblem.pid = tempData.pid;
+            tempCompProblem.shortName = tempData.shortname;
+            competition.compProblems[key] = tempCompProblem;
+          }
+
+          // Parse general competition data
+          competition.cid = data.competition.cid;
+          competition.closed = data.competition.closed;
+          competition.length = data.competition.length;
+          competition.name = data.competition.name;
+          competition.registered = data.competition.registered;
+          competition.startTime = data.competition.startTime;
+
+          // Parse team data
+          for (var i = 0; i < data.teams.length; i++) {
+            var tempTeam = data.teams[i];
+            var tempTeamData = new CompetitionTeam();
+            tempTeamData.displayNames = tempTeam.display_names;
+            tempTeamData.name = tempTeam.name;
+            tempTeamData.problemData = tempTeam.problemData;
+            tempTeamData.users = tempTeam.users;
+            competition.teams.push(tempTeamData);
+          }
+          resolve(competition);
+        } else {
+          resolve(new Competition());
+        }
+      }, (err: Response) => {
+        console.log('Failed to get the competition data!');
+        resolve(new Competition());
+      });
+    });
+  };
+
+  updateCompetition(cid: number, name: string, start: number, stop: number, closed: boolean, compProblems: CompetitionProblem[]) : Promise<Competition> {
```

Comment:
Maybe add a TODO here and for deleteCompetition, just as an indicator that this is something we may want to implement.